### DISCUSSION TITLE
清除权限菜单缓存没用

### DISCUSSION THREAD

**Discussion by zqb666** [zuihou/zuihou-admin-cloud#220] [02.提问]
接口文档有 /api/authority/role/clearCache 但是代码里没找到

**Comment by zuihou** [comment]
SuperCacheController

**Reply by zqb666** [reply]
<img width="1195" alt="image" src="https://user-images.githubusercontent.com/6092525/201632578-9c06cf54-e016-4ccc-a370-7e0b24366d17.png">调试了 一下 发现删除缓存的方法报错

**Reply by zuihou** [reply]
调了什么方法？ 传了什么参数？ 上下文说清楚嘛... 猜不出你的问题...

代码 改一下试试：
``` 
cacheOps.del(collections.stream().map(i -> new UserMenuCacheKeyBuilder().key(i)).toArray(CacheKey[]::new));
```

**Reply by zqb666** [reply]
调用了 saveRoleAuthority方法，是修改某个角色权限保存的时候调用的

**Reply by zqb666** [reply]
cacheOps.del(collections.stream().map(i -> new UserMenuCacheKeyBuilder().key(i)).toArray(CacheKey[]::new));这个可以 不会报错了

**Reply by zuihou** [reply]
> cacheOps.del(collections.stream().map(i -> new UserMenuCacheKeyBuilder().key(i)).toArray(CacheKey[]::new));这个可以 不会报错了

那可能是你jdk版本低了

**Reply by zqb666** [reply]
用的 jdk1.8.0_241
